app/controllers/sessions_controller.rb

```diff
@@ -0,0 +1,19 @@
+class SessionsController < ApplicationController
+  def new; end
+
+  def create
+    user = User.find_by(email: params[:session][:email].downcase)
+    if user&.authenticate(params[:session][:password])
```

Comment:
toán tử &. ý nghĩa là gì e?

Reply:
> toán tử &. ý nghĩa là gì e?

Dạ, &. là safe navigation operator: sẽ trả về nil nếu một phương thức được gọi trên một nil object. Còn như bình thường thì nil_object.method => throws error.

Reply:
Nếu ko sử dụng &. thì bình thường sẽ viết thế nào e?

Reply:
> Nếu ko sử dụng &. thì bình thường sẽ viết thế nào e?

dạ bình thường thì sẽ viết: object && object.method1 && object.method2 ...